src/System.Diagnostics.Process/src/System/Collections/Specialized/StringDictionaryWrapper.cs

```diff
@@ -27,25 +27,38 @@ public StringDictionaryWrapper(Dictionary<string, string> contents)
 
         public override int Count => _contents.Count;
 
-        public override ICollection Keys => _contents.Keys;
-
-        public override ICollection Values => _contents.Values;
+        public override ICollection Keys => ((IDictionary)_contents).Keys;
+        public override ICollection Values => ((IDictionary)_contents).Values;
 
         public override bool IsSynchronized => false;
 
-        public override object SyncRoot => ((ICollection)_contents).SyncRoot;
-        
-        public override void Add(string key, string value) => _contents.Add(key, value);
+        public override object SyncRoot => _contents.SyncRoot;
+
+        public override void Add(string key, string value)
+        {
+            if (_contents.ContainsKey(key))
+            {
```

Comment:
We do need this as we are calling the DictionaryWrapper.Add and it just overrides the key, on desktop in StringDictionary.Add it calls contents.Add where contents is a Hashtable and Hashtable.Add does throw an ArgumentException when the key is duplicate.